packages/cache-redis/src/services/redis-cache.ts

```diff
@@ -0,0 +1,84 @@
+import { Redis } from "ioredis"
+import {
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  ICacheService,
+} from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
+import { RedisCacheModuleOptions } from "../types"
+
+const DEFAULT_CACHE_TIME = 30 // 30 seconds
+const EXPIRY_MODE = "EX" // "EX" stands for an expiry time in second
+
+type InjectedDependencies = {
+  redisClient: Redis
+}
+
+class RedisCacheService implements ICacheService {
+  protected TTL: number
+  protected readonly redis: Redis
+
+  constructor(
+    { redisClient }: InjectedDependencies,
+    options: RedisCacheModuleOptions = {},
+    moduleDeclaration?: ConfigurableModuleDeclaration
+  ) {
+    this.redis = redisClient
+    this.TTL = options.defaultTTL || DEFAULT_CACHE_TIME
+
+    if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+  }
+  /**
+   * Set a key/value pair to the cache.
+   * It is also possible to manage the ttl through environment variable using CACHE_TTL. If the ttl is 0 it will
+   * act like the value should not be cached at all.
+   * @param key
+   * @param data
+   * @param ttl
+   */
+  async set(
+    key: string,
+    data: Record<string, unknown>,
+    ttl: number = this.TTL
+  ): Promise<void> {
+    await this.redis.set(key, JSON.stringify(data), EXPIRY_MODE, ttl)
+  }
+
+  /**
+   * Retrieve a cached value belonging to the given key.
+   * @param cacheKey
+   */
+  async get<T>(cacheKey: string): Promise<T | null> {
+    try {
+      const cached = await this.redis.get(cacheKey)
+      if (cached) {
+        return JSON.parse(cached)
+      }
+    } catch (err) {
+      await this.redis.del(cacheKey)
+    }
+    return null
+  }
+
+  /**
+   * Invalidate cache for a specific key. a key can be either a specific key or more global such as "ps:*".
```

Comment:
should we also support wildcards "*" on the in-memory cache?

Reply:
I think it's a good idea, but is that something we should "enforce" with the `ICacheInterface` somehow to ensure other caching modules implement that behaviour as well, wdyt?

Reply:
Do you think we should hold off on this for now? We will recommend the Redis cache for production environments - even though the in-memory module might be useful for some cases. And then be very explicit about this in our documentation of the in-memory module.

I don't think we should let complex regex acrobatics stand in the way of merging this soon. What's your take?

Reply:
@olivermrbl sounds good! I will add a ticket to implement this in a follow-up iteration and add the recommendation to the Readme.

Reply:
I think that it is a very good idea to include in another iteration 👍

Reply:
@olivermrbl implemented `*` matching for deletion today so now we have the same functionality in redis and inmemory cache modules

Reply:
@fPolic That's great!